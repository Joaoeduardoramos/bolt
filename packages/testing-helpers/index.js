export { isConnected } from './is-connected';
import {
  render as renderServer,
  renderString as renderStringServer,
  stop as stopServer,
} from '@bolt/twig-renderer';
export { fixture as html } from '@open-wc/testing-helpers';

export async function render(template, data) {
  return await renderServer(template, data, true);
}

export async function renderString(template, data) {
  return await renderStringServer(template, data, false);
}

export { stopServer };

// import {
//   // isConnected,
//   // render,
//   // renderString,
//   // stopServer,
//   html,
// } from '../../../testing-helpers';
