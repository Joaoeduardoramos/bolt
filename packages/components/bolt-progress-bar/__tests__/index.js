import {
  render,
  renderString,
  stop as stopTwigRenderer,
} from '@bolt/twig-renderer';
import { fixture as html } from '@open-wc/testing-helpers';

const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../progress-bar.schema.yml'));
const { tag } = schema.properties;

async function renderTwig(template, data) {
  return await render(template, data, true);
}

async function renderTwigString(template, data) {
  return await renderString(template, data, true);
}

const timeout = 120000;

describe('progress bar', async () => {
  let page;

  afterAll(async () => {
    await stopServer();
  }, timeout);

  beforeEach(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      waitUntil: 'networkidle0',
      timeout: 0,
    });
  }, timeout);

  test('basic progress-bar component renders', async () => {
    const results = await renderTwig(
      '@bolt-components-progress-bar/progress-bar.twig',
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('basic progress-bar component with the global `no-shadow` prop added', async () => {
    const results = await renderTwig(
      '@bolt-components-progress-bar/progress-bar.twig',
      {
        no_shadow: true,
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('progress-bar with outer CSS class via Drupal Attributes', async () => {
    const results = await renderTwig(
      '@bolt-components-progress-bar/progress-bar.twig',
      {
        attributes: {
          class: ['u-bolt-margin-top-medium'],
        },
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Default <bolt-progress-bar> w/o Shadow DOM renders', async function() {
    const renderedComponentHTML = await page.evaluate(() => {
      const component = document.createElement('bolt-progress-bar');
      btn.textContent = 'progress-bar Test';
      document.body.appendChild(btn);
      component.useShadow = false;
      component.updated();
      return component.outerHTML;
    });

    const renderedHTML = await html(renderedComponentHTML);

    const image = await page.screenshot();
    expect(image).toMatchImageSnapshot({
      failureThreshold: '0.01',
      failureThresholdType: 'percent',
    });

    expect(renderedHTML).toMatchSnapshot();
  });

  test('Default <bolt-progress-bar> with Shadow DOM renders', async function() {
    const defaultComponentShadowRoot = await page.evaluate(() => {
      const component = document.createElement('bolt-progress-bar');
      component.textContent = 'progress-bar Component Test -- Shadow Root HTML';
      document.body.appendChild(component);
      component.updated();
      return component.renderRoot.innerHTML;
    });

    const defaultComponentOuter = await page.evaluate(() => {
      const component = document.createElement('bolt-progress-bar');
      component.textContent = 'progress-bar Component Test -- Outer HTML';
      document.body.appendChild(component);
      component.updated();
      return component.outerHTML;
    });

    const renderedShadowDomHTML = await html(defaultComponentShadowRoot);
    const renderedHTML = await html(defaultComponentOuter);

    expect(renderedHTML.textContent).toEqual(
      'progress-bar Component Test -- Outer HTML',
    );

    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot({
      failureThreshold: '0.01',
      failureThresholdType: 'percent',
    });

    expect(renderedShadowDomHTML).toMatchSnapshot();
    expect(renderedHTML).toMatchSnapshot();
  });
});
