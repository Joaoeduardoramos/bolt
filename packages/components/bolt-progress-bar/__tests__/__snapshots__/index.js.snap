// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Bar 100% Complete + Title 1`] = `
<bolt-progress-bar value="100"
                   title="Percent Complete"
>
  <div role="progressbar"
       class="c-bolt-progress-bar is-filled    c-bolt-progress-bar--outside"
       aria-valuemin="0"
       aria-valuemax="100"
       aria-valuenow="100"
       aria-label="Percent Complete"
  >
    <div class="c-bolt-progress-bar__title">
      Percent Complete
    </div>
    <div class="c-bolt-progress-bar__track">
      <span class="c-bolt-progress-bar__fill c-bolt-progress-bar__fill--at-max"
            style="transform: translate3d(100%, 0, 0) translate3d(-100%, 0, 0);"
      >
      </span>
    </div>
    <div class="c-bolt-progress-bar__value">
      100%
    </div>
  </div>
</bolt-progress-bar>
`;

exports[`Progress Bar 100% Complete + Title 2`] = `
<div
  style=""
>
  <!---->
  
      
  <div
    aria-label="Percent Complete"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="100"
    class="c-bolt-progress-bar c-bolt-progress-bar--outside"
    role="progressbar"
    style=""
  >
    
        
              
    <div
      class="c-bolt-progress-bar__title"
      style=""
    >
      <!---->
      Percent Complete
      <!---->
    </div>
    
            
        
    <div
      class="c-bolt-progress-bar__track"
      style=""
    >
      
          
      <span
        class="c-bolt-progress-bar__fill c-bolt-progress-bar__fill--at-max"
        style="transform:translate3d(100%, 0, 0) translate3d(-100%, 0, 0);"
      >
        
          
      </span>
      
        
    </div>
    

        
              
    <div
      class="c-bolt-progress-bar__value"
      style=""
    >
      
                100%
              
    </div>
    
            
      
  </div>
  
    
  <!---->
</div>
`;

exports[`Progress Bar Basic Progress Bar usage 1`] = `
<bolt-progress-bar value="50">
  <div role="progressbar"
       class="c-bolt-progress-bar     c-bolt-progress-bar--outside"
       aria-valuemin="0"
       aria-valuemax="100"
       aria-valuenow="50"
  >
    <div class="c-bolt-progress-bar__track">
      <span class="c-bolt-progress-bar__fill "
            style="transform: translate3d(50%, 0, 0) translate3d(-100%, 0, 0);"
      >
      </span>
    </div>
    <div class="c-bolt-progress-bar__value">
      50%
    </div>
  </div>
</bolt-progress-bar>
`;

exports[`Progress Bar Basic Progress Bar usage 2`] = `
<div
  style=""
>
  <!---->
  
      
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
    class="c-bolt-progress-bar c-bolt-progress-bar--outside"
    role="progressbar"
    style=""
  >
    
        
        
    <div
      class="c-bolt-progress-bar__track"
      style=""
    >
      
          
      <span
        class="c-bolt-progress-bar__fill"
        style="transform:translate3d(50%, 0, 0) translate3d(-100%, 0, 0);"
      >
        
          
      </span>
      
        
    </div>
    

        
              
    <div
      class="c-bolt-progress-bar__value"
      style=""
    >
      
                50%
              
    </div>
    
            
      
  </div>
  
    
  <!---->
</div>
`;

exports[`Progress Bar Custom Label Text 1`] = `
<bolt-progress-bar value="2"
                   min="1"
                   max="6"
                   title="Current Progress"
>
  <div role="progressbar"
       class="c-bolt-progress-bar     c-bolt-progress-bar--outside"
       aria-valuemin="1"
       aria-valuemax="6"
       aria-valuenow="2"
       aria-label="Current Progress"
  >
    <div class="c-bolt-progress-bar__title">
      Current Progress
    </div>
    <div class="c-bolt-progress-bar__track">
      <span class="c-bolt-progress-bar__fill "
            style="transform: translate3d(33.333333333333%, 0, 0) translate3d(-100%, 0, 0);"
      >
      </span>
    </div>
    <div class="c-bolt-progress-bar__value">
      33%
    </div>
  </div>
</bolt-progress-bar>
`;

exports[`Progress Bar Custom Label Text 2`] = `
<div
  style=""
>
  <!---->
  
      
  <div
    aria-label="Current Progress"
    aria-valuemax="6"
    aria-valuemin="1"
    aria-valuenow="2"
    class="c-bolt-progress-bar c-bolt-progress-bar--outside"
    role="progressbar"
    style=""
  >
    
        
              
    <div
      class="c-bolt-progress-bar__title"
      style=""
    >
      <!---->
      Current Progress
      <!---->
    </div>
    
            
        
    <div
      class="c-bolt-progress-bar__track"
      style=""
    >
      
          
      <span
        class="c-bolt-progress-bar__fill"
        style="transform:translate3d(33.33333333333333%, 0, 0) translate3d(-100%, 0, 0);"
      >
        
          
      </span>
      
        
    </div>
    

        
              
    <div
      class="c-bolt-progress-bar__value"
      style=""
    >
      
                33%
              
    </div>
    
            
      
  </div>
  
    
  <!---->
</div>
`;

exports[`Progress Bar Progress Bar w/o Shadow DOM 1`] = `
<bolt-progress-bar no-shadow
                   value="50"
>
  <div role="progressbar"
       class="c-bolt-progress-bar     c-bolt-progress-bar--outside"
       aria-valuemin="0"
       aria-valuemax="100"
       aria-valuenow="50"
  >
    <div class="c-bolt-progress-bar__track">
      <span class="c-bolt-progress-bar__fill "
            style="transform: translate3d(50%, 0, 0) translate3d(-100%, 0, 0);"
      >
      </span>
    </div>
    <div class="c-bolt-progress-bar__value">
      50%
    </div>
  </div>
</bolt-progress-bar>
`;

exports[`Progress Bar Progress Bar w/o Shadow DOM 2`] = `
<div
  style=""
>
  <bolt-progress-bar
    no-shadow=""
    style=""
    value="50"
  >
    <!---->
    
      
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="50"
      class="c-bolt-progress-bar c-bolt-progress-bar--outside"
      role="progressbar"
      style=""
    >
      
        
        
      <div
        class="c-bolt-progress-bar__track"
        style=""
      >
        
          
        <span
          class="c-bolt-progress-bar__fill"
          style="transform:translate3d(50%, 0, 0) translate3d(-100%, 0, 0);"
        >
          
          
        </span>
        
        
      </div>
      

        
              
      <div
        class="c-bolt-progress-bar__value"
        style=""
      >
        
                50%
              
      </div>
      
            
      
    </div>
    
    
    <!---->
  </bolt-progress-bar>
</div>
`;

exports[`Progress Bar valuePosition: none 1`] = `
<bolt-progress-bar value="75"
                   title="Percent Downloaded"
                   value-position="none"
>
  <div role="progressbar"
       class="c-bolt-progress-bar     c-bolt-progress-bar--none"
       aria-valuemin="0"
       aria-valuemax="100"
       aria-valuenow="75"
       aria-label="Percent Downloaded"
  >
    <div class="c-bolt-progress-bar__title">
      Percent Downloaded
    </div>
    <div class="c-bolt-progress-bar__track">
      <span class="c-bolt-progress-bar__fill "
            style="transform: translate3d(75%, 0, 0) translate3d(-100%, 0, 0);"
      >
      </span>
    </div>
  </div>
</bolt-progress-bar>
`;

exports[`Progress Bar valuePosition: none 2`] = `
<div
  style=""
>
  <!---->
  
      
  <div
    aria-label="Percent Downloaded"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="75"
    class="c-bolt-progress-bar c-bolt-progress-bar--none"
    role="progressbar"
    style=""
  >
    
        
              
    <div
      class="c-bolt-progress-bar__title"
      style=""
    >
      <!---->
      Percent Downloaded
      <!---->
    </div>
    
            
        
    <div
      class="c-bolt-progress-bar__track"
      style=""
    >
      
          
      <span
        class="c-bolt-progress-bar__fill"
        style="transform:translate3d(75%, 0, 0) translate3d(-100%, 0, 0);"
      >
        
          
      </span>
      
        
    </div>
    

        
      
  </div>
  
    
  <!---->
</div>
`;

exports[`Progress Bar valuePosition: outside 1`] = `
<bolt-progress-bar value="75"
                   title="Percent Downloaded"
                   value-position="outside"
>
  <div role="progressbar"
       class="c-bolt-progress-bar     c-bolt-progress-bar--outside"
       aria-valuemin="0"
       aria-valuemax="100"
       aria-valuenow="75"
       aria-label="Percent Downloaded"
  >
    <div class="c-bolt-progress-bar__title">
      Percent Downloaded
    </div>
    <div class="c-bolt-progress-bar__track">
      <span class="c-bolt-progress-bar__fill "
            style="transform: translate3d(75%, 0, 0) translate3d(-100%, 0, 0);"
      >
      </span>
    </div>
    <div class="c-bolt-progress-bar__value">
      75%
    </div>
  </div>
</bolt-progress-bar>
`;

exports[`Progress Bar valuePosition: outside 2`] = `
<div
  style=""
>
  <!---->
  
      
  <div
    aria-label="Percent Downloaded"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="75"
    class="c-bolt-progress-bar c-bolt-progress-bar--outside"
    role="progressbar"
    style=""
  >
    
        
              
    <div
      class="c-bolt-progress-bar__title"
      style=""
    >
      <!---->
      Percent Downloaded
      <!---->
    </div>
    
            
        
    <div
      class="c-bolt-progress-bar__track"
      style=""
    >
      
          
      <span
        class="c-bolt-progress-bar__fill"
        style="transform:translate3d(75%, 0, 0) translate3d(-100%, 0, 0);"
      >
        
          
      </span>
      
        
    </div>
    

        
              
    <div
      class="c-bolt-progress-bar__value"
      style=""
    >
      
                75%
              
    </div>
    
            
      
  </div>
  
    
  <!---->
</div>
`;

exports[`Progress Bar valuePosition: top 1`] = `
<bolt-progress-bar value="75"
                   title="Percent Downloaded"
                   value-position="top"
>
  <div role="progressbar"
       class="c-bolt-progress-bar     c-bolt-progress-bar--top"
       aria-valuemin="0"
       aria-valuemax="100"
       aria-valuenow="75"
       aria-label="Percent Downloaded"
  >
    <div class="c-bolt-progress-bar__title">
      Percent Downloaded
    </div>
    <div class="c-bolt-progress-bar__track">
      <span class="c-bolt-progress-bar__fill "
            style="transform: translate3d(75%, 0, 0) translate3d(-100%, 0, 0);"
      >
      </span>
    </div>
    <div class="c-bolt-progress-bar__value">
      75%
    </div>
  </div>
</bolt-progress-bar>
`;

exports[`Progress Bar valuePosition: top 2`] = `
<div
  style=""
>
  <!---->
  
      
  <div
    aria-label="Percent Downloaded"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="75"
    class="c-bolt-progress-bar c-bolt-progress-bar--top"
    role="progressbar"
    style=""
  >
    
        
              
    <div
      class="c-bolt-progress-bar__title"
      style=""
    >
      <!---->
      Percent Downloaded
      <!---->
    </div>
    
            
        
    <div
      class="c-bolt-progress-bar__track"
      style=""
    >
      
          
      <span
        class="c-bolt-progress-bar__fill"
        style="transform:translate3d(75%, 0, 0) translate3d(-100%, 0, 0);"
      >
        
          
      </span>
      
        
    </div>
    

        
              
    <div
      class="c-bolt-progress-bar__value"
      style=""
    >
      
                75%
              
    </div>
    
            
      
  </div>
  
    
  <!---->
</div>
`;
